src/mpid/ch4/generic/gpu/Makefile.mk

```diff
@@ -0,0 +1,13 @@
+##
+## Copyright (C) by Argonne National Laboratory.
+##     See COPYRIGHT in top-level directory.
+##
+
+AM_CPPFLAGS += -I$(top_srcdir)/src/mpid/ch4/generic/gpu
+
+noinst_HEADERS   += src/mpid/ch4/generic/gpu/mpidig_gpu.h       \
+                    src/mpid/ch4/generic/gpu/mpidig_gpu_utils.h \
+                    src/mpid/ch4/generic/gpu/mpidig_gpu_types.h
+
+mpi_core_sources += src/mpid/ch4/generic/gpu/mpidig_gpu_init.c    \
+                    src/mpid/ch4/generic/gpu/mpidig_gpu_globals.c
```

Comment:
Is there a reason why these utilities are in the ch4 layer?  Would we not want to use GPU pipelining for the MPI-level collectives, for example?

Reply:
Itsounds reasonable to move into `MPL`.

Reply:
@pavanbalaji A follow-up question, so you prefer the whole `GPU utility function layer` should be placed into `MPL`?

Reply:
No, I don't think `MPL` is the only option.  `MPIR` is another option.